### PR TITLE
Fix: Promotions should involve PR and feature branch (#3516) 

### PR DESCRIPTION
**Pull request by jessebrennan**
https://github.com/DataBiosphere/azul/issues/3516

### Author

- [x] PR title references issue
- [x] PR title matches issue title (preceded by `Fix: ` for bugs)   <sub>or there is a good reason why they're different</sub>
- [x] Title of main commit references issue
- [x] PR is connected to Zenhub issue and description links to issue

### Author (reindex)

- [x] Added `r` tag to commit title                         <sub>or this PR does not require reindexing</sub>
- [x] Added `reindex` label to PR                           <sub>or this PR does not require reindexing</sub>

### Author (freebies & chains)

- [x] Freebies are blocked on this PR                       <sub>or there are no freebies in this PR</sub>
- [x] Freebies are referenced in commit titles              <sub>or there are no freebies in this PR</sub>
- [x] This PR is blocked by previous PR in the chain        <sub>or this PR is not chained to another PR</sub>
- [x] Added `chain` label to the blocking PR                <sub>or this PR is not chained to another PR</sub>

### Author (upgrading)

- [x] Documented upgrading of deployments in UPGRADING.rst  <sub>or this PR does not require upgrading</sub>
- [x] Added `u` tag to commit title                         <sub>or this PR does not require upgrading</sub>
- [x] Added `upgrade` label to PR                           <sub>or this PR does not require upgrading</sub>
- [x] Added announcement to PR description                  <sub>or this PR does not require announcement</sub>
- [x] Added checklist items for additional operator tasks   <sub>or this PR does not require additional tasks</sub>

### Author (requirements, before every review)

- [x] Ran `make requirements_update`                        <sub>or this PR does not touch requirements*.txt, common.mk, Makefile and Dockerfile</sub>
- [x] Added `R` tag to commit title                         <sub>or this PR does not touch requirements*.txt</sub>
- [x] Added `reqs` label to PR                              <sub>or this PR does not touch requirements*.txt</sub>

### Author (before every review)

- [x] `make integration_test` passes in personal deployment <sub>or this PR does not touch functionality that could break the IT</sub>
- [x] Rebased branch on `develop`, squashed old fixups

### Primary reviewer (after approval)

- [x] Commented in issue about demo expectations            <sub>or labelled issue as `no demo`</sub>
- [x] Decided if PR can be labeled `no sandbox`
- [x] PR title is appropriate as title of merge commit
- [x] Moved ticket to Approved column
- [x] Assigned PR to an operator

### Operator (before pushing merge the commit)

- [x] Checked `reindex` label and `r` commit title tag
- [x] Checked that demo expectations are clear              <sub>or issue is labeled as `no demo`</sub>
- [x] Rebased and squashed branch
- [x] Sanity-checked history
- [x] Pushed PR branch to Github
- [x] Branch pushed to Gitlab and added `sandbox` label     <sub>or PR is labeled `no sandbox`</sub>
- [x] Build passed in sandbox                               <sub>or PR is labeled `no sandbox`</sub>
- [x] Started reindex in `sandbox`                          <sub>or this PR does not require reindexing `sandbox`</sub>
- [x] Checked for failures in `sandbox`                     <sub>or this PR does not require reindexing `sandbox`</sub>
- [x] Added PR reference to merge commit title
- [x] Collected commit title tags in merge commit title
- [ ] Moved linked issue to Merged column
- [ ] Pushed merge commit to Github

### Operator (after pushing the merge commit)

- [ ] Made announcement requested by author                 <sub>or PR description does not contain an announcement</sub>
- [ ] Moved freebies to Merged column                       <sub>or there are no freebies in this PR</sub> 
- [ ] Shortened the PR chain                                <sub>or this PR is not the base of another PR</sub>
- [ ] Verified that `N reviews` labelling is accurate       <sub>or or this PR is authored by lead</sub>
- [ ] Pushed merge commit to Gitlab                         <sub>or or merge commit can be pushed later, with another PR</sub>
- [ ] Deleted PR branch from Github and Gitlab
- [ ] Build passes on Gitlab

### Operator (reindex) 

- [ ] Started reindex in target deployment                  <sub>or this PR does not require reindexing</sub>
- [ ] Checked for and triaged indexing failures             <sub>or this PR does not require reindexing</sub>
- [ ] Emptied fail queues in target deployment              <sub>or this PR does not require reindexing</sub>

### Operator

- [ ] Unassigned PR

### CODE REVIEW

File: OPERATOR.rst [hunk 1]
```diff
@@ -211,28 +211,17 @@ To do a promotion:
...
 
-#. ::
-
-      git merge develop --no-ff
-
-   Use the default merge commit message, which should be ``Merge branch 'develop' into prod``.
-
-#. Examine the changes to ``prod`` since the last promotion to determine if a reindex
-   is necessary.
+#. File a PR on github from the new promotions branch which targets ``prod``.
```
Comment (achave11) [inline OPERATOR.rst:218]:
```suggestion
#. File a PR on Github from the new promotions branch which targets ``prod``.
```

**Review by achave11 — APPROVED** [review]
One observation to be addressed. Otherwise, it looks good!

File: OPERATOR.rst (hunk 1 above)
Comment (noah-aviel-dove) [reply OPERATOR.rst]: Thanks for spotting that. I think this review probably shouldn't have been an approval though.

**Comment by noah-aviel-dove** [comment]
Build is red due to https://github.com/DataBiosphere/azul/issues/3603

**Review by achave11 — APPROVED** [review]
LGTM! 👍🏽

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]
Build not green.

**Comment by hannes-ucsc** [comment]
Saw the comment just now. Still won't review until build is green.

File: OPERATOR.rst [hunk 2]
```diff
@@ -211,28 +211,17 @@ To do a promotion:
...
 
-#. ::
-
-      git merge develop --no-ff
-
-   Use the default merge commit message, which should be ``Merge branch 'develop' into prod``.
-
-#. Examine the changes to ``prod`` since the last promotion to determine if a reindex
-   is necessary.
+#. File a PR on GitHub from the new promotions branch which targets ``prod``.
```
Comment (hannes-ucsc) [inline OPERATOR.rst:218]:
```suggestion
#. File a PR on GitHub from the new promotions branch. The PR must target ``prod``.
```

File: OPERATOR.rst [hunk 3]
```diff
@@ -211,28 +211,17 @@ To do a promotion:
 
 #. ::
 
-      git checkout prod
+      git checkout develop
```
Comment (hannes-ucsc) [inline OPERATOR.rst:214]:
```suggestion
      git checkout develop
      git pull -ff-only
```

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]

File: OPERATOR.rst [hunk 4]
```diff
@@ -211,28 +211,18 @@ To do a promotion:
...
-#. ::
-
-       git push github
-
-#. ::
-
-       git push gitlab.dcp2.prod
-
-#. Monitor reindex and check / triage any failures.
+#. When merging, follow the checklist and making sure to carry over any commit
```
Comment (hannes-ucsc) [inline OPERATOR.rst:223]:
```suggestion
#. When merging, follow the checklist and make sure to carry over any commit
```

**Review by hannes-ucsc — APPROVED** [review]

**Pull request merged**